Backend/Remora.Discord.API/API/CDN.cs

```diff
@@ -643,7 +643,7 @@ public static Result<Uri> GetApplicationCoverUrl
             Optional<ushort> imageSize = default
         )
         {
-            if (!application.CoverImage.HasValue)
+            if (!application.CoverImage.IsSpecified || application.CoverImage.Value is null)
```

Comment:
And here we have a great example of the other half of this problem: `.CoverImage` here is defined as `Optional<IImageHash>`. In reality, there is no possibility for this to be `null`, and `Optional<T>` shouldn't introduce one. I.E. `Optional<T>.Value` should be `T` not `T?`.

Reply:
I've restored some things temporarily-

It seems `Optional.HasValue` is used in place of nullability checks in cases just like this all around the code.